plugins/cache-ehcache/src/main/java/org/opensearch/cache/EhcacheDiskCacheSettings.java

```diff
@@ -36,17 +37,25 @@ public class EhcacheDiskCacheSettings {
 
     public static final Setting.AffixSetting<Integer> DISK_WRITE_MINIMUM_THREADS_SETTING = Setting.suffixKeySetting(
         EhcacheDiskCache.EhcacheDiskCacheFactory.EHCACHE_DISK_CACHE_NAME + ".min_threads",
-        (key) -> Setting.intSetting(key, 2, 1, 5, NodeScope)
+        (key) -> Setting.intSetting(key, 2, 1, max(2, Runtime.getRuntime().availableProcessors() / 8), NodeScope)
     );
 
     /**
-     *  Ehcache disk write maximum threads for its pool
+     *  Ehcache disk write maximum threads for its pool. We max disk write threads to ony go upto max(2, N / 8) where N
+     *  is the number of CPU cores. This is done as disk operations are typically I/O bound rather than CPU bound, so
+     *  can't scale this blindly just based on the CPU cores.
      *
      *  Setting pattern: {cache_type}.ehcache_disk.max_threads
      */
     public static final Setting.AffixSetting<Integer> DISK_WRITE_MAXIMUM_THREADS_SETTING = Setting.suffixKeySetting(
         EhcacheDiskCache.EhcacheDiskCacheFactory.EHCACHE_DISK_CACHE_NAME + ".max_threads",
-        (key) -> Setting.intSetting(key, 2, 1, 20, NodeScope)
+        (key) -> Setting.intSetting(
+            key,
+            max(2, Runtime.getRuntime().availableProcessors() / 8),
+            1,
+            Runtime.getRuntime().availableProcessors(),
+            NodeScope
+        )
```

Comment:
Reducing the number of threads can cause queuing of disk write operations, unless there is some batching logic within Ehcache. IMO, it is desirable for TieredCaching to write the computed results to disk as soon as possible. Number of threads is not worrisome as long as those threads are not compute intensive, which in this case they are not

Reply:
Increased default threads to number of search threads.

Reply:
If we dynamically change the number of search threads, how will that impact disk write threads?

Reply:
Disk write threads would still remain in that case, and would need one to change the static setting. I am not sure if there is a way as of now to change search threads dynamically? 

Also in addition to this, this is a generic setting ie it can be used for non search usecases as well in the future, pinning it against search threads only makes sense for requestCache. So I think we should define the default for this setting independently wrt cpu cores, (cpu * 1.5) looks fine to me for now.